docs/design.md

```diff
@@ -287,6 +287,11 @@ Such that the `Foo` field is a map, keyed on the type of the key leaf (`fookey`)
 
 Each YANG list that exists within a container has a helper-method generated for it. For a list named `foo`, the parent container (`C`) has a `NewFoo(fookey string)` method generated, taking a key value as an argument, and returning a new member of the map within the `foo` list.
 
+##### Note on using binary as a list key type
+Because `Binary`'s underlying `[]byte` type is not hashable, YANG models
+containing lists with `binary` as a key value, or a `union` type containing a
+`binary` type is not supported by ygot, and will not compile.
```

Comment:
Perhaps state here `an error is returned by the code generation process for such cases, this is a known limitation`.